### PR TITLE
Sync with gfx-rs link_program

### PR DESCRIPTION
**Pull request by mninja**
Syncs the code with this PR https://github.com/gfx-rs/gfx-rs/pull/657

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -21,7 +21,7 @@
 #[macro_use]
 extern crate log;
 extern crate libc;
-extern crate gfx_gl as gl;
+extern crate "gfx_gl" as gl;
```
Comment (kvark) [inline src/lib.rs]: Isn't this triggering a cargo warning? I thought the quotes are thing of the past.

File: src/shade.rs [hunk 2]
```diff
@@ -311,24 +311,36 @@ pub fn create_program<I: Iterator<Item = super::Shader>>(gl: &gl::Gl,
...
-        unsafe {
-            gl.GetProgramInfoLog(name, length, &mut length,
-                (&log[..]).as_ptr() as *mut gl::types::GLchar);
+
+    // fetch the log if the buffer was provided
+    match log {
+        Some(log) => {
+            let mut length  = get_program_iv(gl, name, gl::INFO_LOG_LENGTH);
+            if length > 0 {
+                log.reserve(length as usize);
```
Comment (kvark) [inline src/shade.rs]: Shouldn't we clear it first? reserving allocates in addition to the current size, IIRC.

File: src/lib.rs (hunk 1 above)
Comment (bjz) [inline src/lib.rs]: Yeah we need to get rid of the quotes.

**Comment by kvark** [comment]
Closing due to inactivity. Please feel free to reopen.

**Pull request closed without merging**